homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -296,61 +326,103 @@ def parse_hue_api_put_light_body(request_json, entity):
+        device_type = HUE_API_DEVICE_TYPE_DIMMABLE
 
-        # Make sure the entity actually supports brightness
-        entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+        state = {HUE_API_STATE_ON: is_on}
+        brightness = entity.attributes.get(ATTR_BRIGHTNESS,
+                                           255 if is_on else 0)
+        state[HUE_API_STATE_BRI] = brightness
 
         if entity.domain == "light":
-            if entity_features & SUPPORT_BRIGHTNESS:
-                pass
+            # Make sure the entity actually supports brightness
+            entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+            if entity_features & SUPPORT_COLOR_TEMP:
+                color_temp = entity.attributes.get(ATTR_COLOR_TEMP, 0)
+                state[HUE_API_STATE_CT] = color_temp
+
+            if entity_features & SUPPORT_XY_COLOR:
+                xy_color = entity.attributes.get(ATTR_XY_COLOR, [0.0, 0.0])
+                state[HUE_API_STATE_XY] = xy_color
+            elif entity_features & SUPPORT_RGB_COLOR:
+                rgb_color = entity.attributes.get(ATTR_RGB_COLOR, [0, 0, 0])
+                xy_color = color_util.color_RGB_to_xy(
+                    *(int(val) for val in rgb_color))
+                state[HUE_API_STATE_XY] = xy_color
+
+            if entity_features & SUPPORT_XY_COLOR or \
+                    entity_features & SUPPORT_RGB_COLOR:
+                state[HUE_API_STATE_COLORMODE] = HUE_API_STATE_XY
+                if entity_features & SUPPORT_COLOR_TEMP:
+                    device_type = HUE_API_DEVICE_TYPE_EXTENDED_COLOR
+                else:
+                    device_type = HUE_API_DEVICE_TYPE_COLOR
+            elif entity_features & SUPPORT_COLOR_TEMP:
+                state[HUE_API_STATE_COLORMODE] = HUE_API_STATE_CT
+                device_type = HUE_API_DEVICE_TYPE_COLOR_TEMP
 
         elif entity.domain == "media_player":
             level = entity.attributes.get(
-                ATTR_MEDIA_VOLUME_LEVEL, 1.0 if final_state else 0.0)
+                ATTR_MEDIA_VOLUME_LEVEL, 1.0 if is_on else 0.0)
             # Convert 0.0-1.0 to 0-255
-            final_brightness = round(min(1.0, level) * 255)
+            state[HUE_API_STATE_BRI] = round(min(1.0, level) * 255)
+
         elif entity.domain == "fan":
             speed = entity.attributes.get(ATTR_SPEED, 0)
             # Convert 0.0-1.0 to 0-255
-            final_brightness = 0
+            state[HUE_API_STATE_BRI] = 0
             if speed == SPEED_LOW:
-                final_brightness = 85
+                state[HUE_API_STATE_BRI] = 85
             elif speed == SPEED_MEDIUM:
-                final_brightness = 170
+                state[HUE_API_STATE_BRI] = 170
             elif speed == SPEED_HIGH:
-                final_brightness = 255
+                state[HUE_API_STATE_BRI] = 255
+
     else:
+        device_type = HUE_API_DEVICE_TYPE_DIMMABLE
```

Comment:
The problem is that we don't know the device type in the put request (where state is cached). I was hoping to cache it initially. I'll put it in the common path.